dotnet-platform-specific/RetrievalOfCpuCores/RetrievalOfCpuCores/CpuInformation.cs

```diff
@@ -0,0 +1,121 @@
+    /// </summary>
+    /// <returns>number of logical processors</returns>
+    public int GetProcessorsCount()
+    {
+        return Environment.ProcessorCount;
+    }
+
+    /// <summary>
+    /// Retrieves the number of logical processors of the CPU.
+    /// </summary>
+    /// <returns>number of logical processors</returns>
+    public int GetLogicalProcessors()
+    {
+        foreach (var item in new System.Management.ManagementObjectSearcher("Select * from Win32_ComputerSystem").Get())
+        {
+            return int.Parse(item["NumberOfLogicalProcessors"].ToString());
+        }
+        return 0;
+    }
+
+    /// <summary>
+    /// Retrieves the number of cores of the CPU.
+    /// This is only supported on windows.
+    /// </summary>
+    /// <returns>number of cores of the CPU</returns>
+    public int GetNumberOfCores()
+    {
+        var coreCount = 0;
+        foreach (var item in new System.Management.ManagementObjectSearcher("Select * from Win32_Processor").Get())
+        {
+            coreCount += int.Parse(item["NumberOfCores"].ToString());
+        }
+        return coreCount;
+    }
+
+    /// <summary>
+    /// Retrieves the number of physical processors of the CPU.
+    /// This is only supported on windows.
+    /// </summary>
+    /// <returns>number of physical processors</returns>
+    public int GetPhysicalProcessors()
+    {
+        foreach (var item in new System.Management.ManagementObjectSearcher("Select * from Win32_ComputerSystem").Get())
+        {
+            return int.Parse(item["NumberOfProcessors"].ToString());
+        }
+        return 0;
+    }
+
+    /// <summary>
+    /// Retrieves the number of logical processors of the CPU, including also the excluded processors.
+    /// </summary>
+    /// <returns></returns>
+    public int GetExcludedProcessors()
+    {
+        var deviceCount = 0;
+        var deviceList = IntPtr.Zero;
+        var processorGuid = new Guid("{50127dc3-0f36-415e-a6cc-4cb3be910b65}");
+
+        try
+        {
+            deviceList = SetupDiGetClassDevs(ref processorGuid, "ACPI", IntPtr.Zero, (int)DIGCF.PRESENT);
+            for (var deviceNumber = 0; ; deviceNumber++)
+            {
+                var deviceInfo = new SP_DEVINFO_DATA();
+                deviceInfo.cbSize = Marshal.SizeOf(deviceInfo);
+
+                if (!SetupDiEnumDeviceInfo(deviceList, deviceNumber, ref deviceInfo))
+                {
+                    deviceCount = deviceNumber;
+                    break;
+                }
+            }
+        }
+        finally
+        {
+            if (deviceList != IntPtr.Zero) { SetupDiDestroyDeviceInfoList(deviceList); }
+        }
+        return deviceCount;
+    }
+
+    [DllImport("setupapi.dll", SetLastError = true)]
+    private static extern IntPtr SetupDiGetClassDevs(ref Guid ClassGuid,
+        [MarshalAs(UnmanagedType.LPStr)] String enumerator,
```

Comment:
@lennart02 
Let's use `string` instead of `String` and `int` instead of `Int32`